shared/provision/set-public-name/index.tsx

```diff
@@ -19,6 +19,15 @@ type Props = {
 const SetPublicName = (props: Props) => {
   const [deviceName, setDeviceName] = React.useState(defaultDevicename)
   const cleanDeviceName = Constants.cleanDeviceName(deviceName)
+  const disabled = React.useMemo(() => {
```

Comment:
is there an advantage to memoing this as this'll get called as you're typing and you'll only skip if somehow this rerenders w/o it being due to the devicename? also likely not expensive to just calc it